.github/workflows/.reusable-docker-build.yml

```diff
@@ -67,18 +76,38 @@ jobs:
             type=ref,event=branch
             type=ref,event=tag
             type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
 
       - name: Build and push image
         uses: depot/build-push-action@v1
         with:
+          context: .
           push: true
-          build-args: ${{ inputs.build-args }}
+          platforms: linux/amd64,linux/arm64
+          secrets: ${{ secrets.secrets }}
+          target: ${{ inputs.target }}
+          build-args: |
+            CI_COMMIT_SHA=${{ github.sha }}
+            ${{ inputs.build-args }}
           file: ${{ inputs.file }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
 
       - name: Run Trivy vulnerability scanner
-        if: ${{ inputs.scan }}
+        id: trivy
+        if: inputs.scan
         uses: aquasecurity/trivy-action@master
         with:
           image-ref: ${{ inputs.registry-url }}/flagsmith/${{ inputs.image-name }}:${{ steps.meta.outputs.version }}
+          format: sarif
+          output: trivy-results.sarif
+        env:
+          TRIVY_USERNAME: ${{ github.actor }}
+          TRIVY_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        if: inputs.scan && (success() || failure())
```

Comment:
what other options are there other than `success()` and `failure()` here? `error()` maybe I guess?

Reply:
This condition prompts the results upload step to happend even if `trivy-action` fails (but not if `scan` argument is `false`). `trivy-action` might fail if scan results cross a certain threshold.

Reply:
My question though was ... what other statuses are there? `success` and `failure` seem to cover all the bases so isn't this statement just the same as `if: inputs.scan` ?

Reply:
No, the default behaviour is equivalent to `if: inputs.scan && success()`.

Overall, besides `success()` and `failure()` there's also `skipped()`, which we don't want to ignore.